ui/app/AppLayouts/Chat/stores/MessageStore.qml

```diff
@@ -11,13 +11,49 @@ QtObject {
     readonly property bool loadingHistoryMessagesInProgress: root.chatSectionModule? root.chatSectionModule.loadingHistoryMessagesInProgress : false
     readonly property int newMessagesCount: messagesModel ? messagesModel.newMessagesCount : 0
     readonly property bool messageSearchOngoing: messageModule ? messageModule.messageSearchOngoing : false
+    readonly property bool amIChatAdmin: d.amIChatAdmin
+    readonly property bool isPinAllowedForAllMembers: d.isPinAllowedForAllMembers
+    readonly property string chatId: d.chatId
+    readonly property int chatType: d.chatType
+    readonly property string chatColor: d.chatColor
+    readonly property string chatIcon: d.chatIcon
 
     onMessageModuleChanged: {
         if(!messageModule)
             return
         root.messagesModel = messageModule.model
     }
 
+    readonly property QtObject d: QtObject {
+        //These properties are used to avoid calling the nim functions too often
+        property bool amIChatAdmin: messageModule ? messageModule.amIChatAdmin() : false
+        property bool isPinAllowedForAllMembers: messageModule ? messageModule.pinMessageAllowedForMembers() : false
+        property string chatId: messageModule ? messageModule.getChatId() : ""
+        property int chatType: messageModule ? messageModule.getChatType() : 0
+        property string chatColor: messageModule ? messageModule.getChatColor() : ""
+        property string chatIcon: messageModule ? messageModule.getChatIcon() : ""
```

Comment:
all above `readonly`?

Reply:
The properties are updated by refreshData. This is why I haven't used readonly

Reply:
I see